lib/src/test/java/io/ably/lib/test/realtime/ConnectionManagerTest.java

```diff
@@ -388,4 +389,32 @@ public void run() {
 			fail("init0: Unexpected exception instantiating library");
 		}
 	}
+
+	/**
+	 * Connect and then verify that the connection manager has the default value for connectionStateTtl;
+	 */
+	@Test
+	public void connection_details_has_ttl() {
+		try {
+			ClientOptions opts = createOptions(testVars.keys[0].keyStr);
+			final AblyRealtime ably = new AblyRealtime(opts);
+			ably.connection.on(ConnectionEvent.connected, new ConnectionStateListener() {
+				@Override
+				public void onConnectionStateChanged(ConnectionStateChange state) {
+					assertEquals(1, 1);
+					try {
+						Field field = ably.connection.connectionManager.getClass().getDeclaredField("connectionStateTtl");
+						field.setAccessible(true);
+						assertEquals("Verify connectionStateTtl has the default value", field.get(ably.connection.connectionManager), 120000L);
+					} catch (NoSuchFieldException|IllegalAccessException e) {
+						fail("Unexpected exception in checking connectionStateTtl");
+					}
+					ably.close();
+				}
+			});
```

Comment:
There is no wait as far as I can tell so the `onConnectionStateChanged` is not necessarily waited for?  How do you know this test even run and assertion succeeded?

Reply:
@mattheworiordan I made the test more explicit https://github.com/ably/ably-java/pull/389/commits/850eaf9d2a7f186bcd73ad1594903142eccd0086